### ISSUE TITLE
Code review

### ISSUE THREAD

**Issue by aquastripe** [Rita012-cyber/Plant_Seedlings_Classification#1]
- 取出預測結果的方式
  ```py
  _, preds = torch.max(outputs, 1)
  ```
  改成
  ```py
  preds = torch.argmax(outputs, 1)
  ```
  會更直白、簡潔
- ```py
  image = transforms.Resize((224, 224))(to_pil_image(image))
  ```
  這樣寫有些多此一舉，如果你用 `torchvision.io.read_image` 你應該使用 `torchvision.transforms.v2` 的 API
  這樣你的 transforms 不需要再轉到 PIL 才能使用
- 這份實作考慮 resumable 很不錯，但是沒有考慮到在 resume 的過程要對 dataset random splitting 設定一個 random seed 使得每次 resume 的時候都是用相同的方式分割資料集。這可能也是你第 23 epoch 時 validation loss 驟降的原因 (混進 training set)。最理想的情況是 training loss < validation loss 除了訓練初期 (第一個 epoch)。
